spirv_msl.cpp

```diff
@@ -334,6 +334,20 @@ void CompilerMSL::emit_entry_point_declarations()
 		          type.basetype == SPIRType::SampledImage ? to_sampler_expression(samp.first) : to_name(samp.first),
 		          "(", merge(args), ");");
 	}
+
+	// Emit buffer arrays here.
+	for (uint32_t array_id : buffer_arrays)
+	{
+		const auto &var = get<SPIRVariable>(array_id);
+		const auto &type = get<SPIRType>(var.basetype);
+		string name = get_name(array_id);
+		statement(get_argument_address_space(var) + " " + type_to_glsl(type) + "* " + name + "[] =");
+		begin_scope();
+		for (uint32_t i = 0; i < type.array[0]; ++i)
+			statement(name + "_" + convert_to_string(i) + ",");
+		end_scope_decl();
```

Comment:
Probably need to emit a blank line here after declaring the array for cleaner output, the declarations are smashed together.